odoo/modules/neutralize.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import textwrap
+
+import odoo
+from .module import get_manifest
+
+
+def get_installed_modules(cursor):
+    cursor.execute('''
+        SELECT name FROM ir_module_module
+        WHERE state IN ('installed', 'to upgrade', 'to remove', 'to install');
+    ''')
+    return [result[0] for result in cursor.fetchall()]
+
+def get_neutralization_sql(modules):
+    queries = []
+
+    # neutralization for each module
+    for module in modules:
+        manifest = get_manifest(module)
+        if manifest and manifest['neutralize']:
+            f = odoo.modules.get_module_resource(module, manifest['neutralize'])
+            with odoo.tools.misc.file_open(f) as neutralize_sql:
+                queries.append(neutralize_sql.read())
+
+    # neutralization flags for the database
+    queries.append(textwrap.dedent('''INSERT INTO ir_config_parameter(key, value)
+        VALUES ('database.is_neutralized', true)
+        ON CONFLICT (key)
+        DO UPDATE SET value = true;
+    '''))
+
+    # neutralization watermarks
+    queries.append(textwrap.dedent('''UPDATE ir_ui_view
+        SET active=true
+        WHERE key IN ('web.neutralize_banner', 'website.neutralize_ribbon');
+    '''))
+
+    return '\n\n'.join(queries)
```

Comment:
This is slightly unsafe, as it _requires_ all neutralization queries _to end with_ a semicolon.  The common practice is to use the semicolon as a query _separator_.  Should we add semicolon into the separator string to make it safer?  I don't know what's best in this case.

Reply:
I theoretically agree about the safety point but, in practice, the neutralization queries will only be written by developers (on Odoo and other platforms), so we want to enforce these queries to be properly written (with semicolon in this case). To ensure this, we have the `test_neutralize.py` test file and as long as it is ran, we won't have issues.
For platforms like OdooSH, it makes sense for the platforms to keep this test as it may englobe tests for customized modules.

Reply:
@rco-odoo yeah I totally understand your point but I second what Laurent says, and not only for Odoo.sh.

We want to allow on-premise users to create a test dump by themselves if they want to, eg:

```bash
$ pg_dump -d foobar > out.dump
$ odoo-bin neutralize -d foobar --stdout >> out.dump
```

However if you have a better idea to accomplish the same goal please share !